e2e/nomostest/portforwarder/port_forwarder.go

```diff
@@ -183,74 +235,92 @@ func (pf *PortForwarder) portForwardToPod(pod string, async bool) error {
 		// Forwarding from 127.0.0.1:44043
 		_, err = fmt.Sscanf(line, "Forwarding from 127.0.0.1:%d", &localPort)
 		if err != nil {
-			return fmt.Errorf("unable to parse port-forward output: %q", s)
+			return errors.Errorf("unable to parse port-forward output: %q", s)
 		}
 		return nil
 	})
 	if err != nil {
-		return err
+		return nil, err
 	}
+	// We can't stop cmd.Process.Wait(), but we can at least consume the error
+	// when it's done, to let its goroutine return.
+	// The caller can start their own wait, if they want.
+	go func() {
+		<-exitCh
```

Comment:
It looks like this error just gets swallowed, what is the purpose of this and the goroutine on line 206 which captures the error?

Reply:
`portForwardToPod` only cares about premature exit now. It watches until the retry loop is done, and then doesn't care about any exit signal after that, but there's no way to stop the `cmd.Process.Wait()` call. So it spins up another goroutine to consume the error or nil from the channel, to allow both goroutines to return.

After `portForwardToPod` returns and `portForwardToDeployment` returns, `start` calls `updateSubprocess` which calls `waitAndUpdateSubprocess` in a goroutine. `waitAndUpdateSubprocess` calls `proc.cmd.Wait()` and blocks until the process has exited.

This complexity allows `portForwardToPod` to be ignorant of what happens to the process after it returns. It doesn't need to know how to update the subprocess on the PortForwarder, and it doesn't need to know how to run async or lock. It just returns and lets the caller figure that out.